src/semgrep_agent/main.py

```diff
@@ -150,14 +152,30 @@ def main(
 
     committed_datetime = meta.commit.committed_datetime if meta.commit else None
 
-    results = semgrep.scan(
-        config,
-        committed_datetime,
-        meta.base_commit_ref,
-        meta.head_ref,
-        semgrep.get_semgrepignore(sapp.scan.ignore_patterns),
-        sapp.is_configured,
-    )
+    try:
+        results = semgrep.scan(
+            config,
+            committed_datetime,
+            meta.base_commit_ref,
+            meta.head_ref,
+            semgrep.get_semgrepignore(sapp.scan.ignore_patterns),
+            sapp.is_configured,
+        )
+    except SemgrepError as error:
+        print_sh_error_info(error.stdout, error.stderr, error.command, error.exit_code)
+
+        # If logged in handle exception
+        if sapp.is_configured:
+            exit_code = sapp.report_failure(error)
+            if exit_code == 0:
+                click.echo(
+                    f"Was going to exit with code {error.exit_code} but semgrep-app configuration is set to fail open so exiting with code 0.",
```

Comment:
```suggestion
                    f"Semgrep returned an error (code {error.exit_code}). However, this project's policy on {sapp.url} is configured to pass the build on Semgrep errors. This action will exit with return code 0.",
```

Reply:
I'll create a follow up PR with this change